wakuv2/waku_test.go

```diff
@@ -280,3 +282,103 @@ func TestWakuV2Filter(t *testing.T) {
 
 	require.NoError(t, w.Stop())
 }
+
+func TestWakuV2Store(t *testing.T) {
+	// Configuration for the first Waku node
+	sql1, err := helpers.SetupTestMemorySQLDB(appdatabase.DbInitializer{})
+	require.NoError(t, err)
+	config1 := &Config{
+		Port:              0,
+		EnableDiscV5:      false,
+		DiscoveryLimit:    20,
+		EnableStore:       true,
+		StoreCapacity:     100,
+		StoreSeconds:      3600,
+		KeepAliveInterval: 10,
+	}
+
+	// Start the first Waku node
+	w1, err := New("", "", config1, nil, sql1, nil, nil, nil)
+	require.NoError(t, err)
+	require.NoError(t, w1.Start())
+	defer func() {
+		require.NoError(t, w1.Stop())
+	}()
+
+	// Configuration for the second Waku node
+	sql2, err := helpers.SetupTestMemorySQLDB(appdatabase.DbInitializer{})
+	require.NoError(t, err)
+	config2 := &Config{
+		Port:              0,
+		EnableDiscV5:      false,
+		DiscoveryLimit:    20,
+		EnableStore:       true,
+		StoreCapacity:     100,
+		StoreSeconds:      3600,
+		KeepAliveInterval: 10,
+	}
+
+	// Start the second Waku node
+	w2, err := New("", "", config2, nil, sql2, nil, nil, nil)
+	require.NoError(t, err)
+	require.NoError(t, w2.Start())
+	defer func() {
+		require.NoError(t, w2.Stop())
+	}()
+
+	// Connect the two nodes directly
+	peer2Addr := w2.node.ListenAddresses()[0].String()
+	err = w1.node.DialPeer(context.Background(), peer2Addr)
+	require.NoError(t, err)
+
+	// Sanity check, not great, but it's probably helpful
+	options := func(b *backoff.ExponentialBackOff) {
+		b.MaxElapsedTime = 30 * time.Second
+	}
+	err = tt.RetryWithBackOff(func() error {
+		if len(w1.Peers()) == 0 {
+			return errors.New("no peers discovered")
+		}
+		return nil
+	}, options)
+	require.NoError(t, err)
+
+	// Wait for the nodes to discover each other
+	time.Sleep(1 * time.Second)
```

Comment:
We use `Sleep` and `RetryWithBackOff` to wait for peers for discovered. I think this is often used in other tests too, however, just realised that we have this: https://github.com/status-im/status-go/blob/8e7d0a523e803f5f3b62f9b04deb3dd67449da16/wakuv2/waku.go#L1192
One can add a connection status subscription via `Waku.SubscribeToConnStatusChanges` and wait for `ConnStatus` structs to arrive.

Reply:
awesome, I'll try that